web/geosearch/datapunt_geosearch/datasource.py

```diff
@@ -131,41 +132,54 @@ def execute_queries(self, datasets=None):
     def execute_point_query(self, cur, table, temporal_bounds=None):
+        tmp_bounds = sql.SQL("")
+        if temporal_bounds is not None:
+            start, end = temporal_bounds
+            tmp_bounds = sql.SQL(
+                " AND ({start} < now() or {start} IS NULL) AND ({end} > now() OR {end} IS NULL)"
+            ).format(start=sql.Identifier(start), end=sql.Identifier(end))
+
         if not self.use_rd:
             # In this case, coordinates are assumed to be latlng and projected to rijksdriehoek
-            sql = """
-                SELECT {},
-                    ST_Distance({},
-                    ST_Transform(ST_GeomFromText('POINT(%s %s)', 4326), 28992)) AS distance
-                FROM {}
-                WHERE ST_DWithin({}, ST_Transform(ST_GeomFromText('POINT(%s %s)', 4326), 28992), %s) {}
-            """.format(
-                self.fields, self.meta["geofield"], table, self.meta["geofield"], self.extra_where
+            stmt = sql.SQL("""
+                SELECT {fields},
+                    ST_Distance({geo_field},
+                    ST_Transform(ST_GeomFromText('POINT({y} {x})', 4326), 28992)) AS distance
+                FROM {schema}.{table_name}
+                WHERE ST_DWithin({geo_field}, ST_Transform(ST_GeomFromText('POINT({y} {x})', 4326), 28992), {radius}) {extra_where} {temp_predicate} ORDER BY distance {limit}
+            """).format(
+                fields=sql.SQL(", ").join(map(sql.SQL, self.fields.split(","))), # Properly escape SQL in self.fields
+                table_name=sql.Identifier(table.split(".")[1]),
+                schema=sql.Identifier(table.split(".")[0]),
+                geo_field=sql.Identifier(self.meta["geofield"]),
+                x=sql.Placeholder("x"),
+                y=sql.Placeholder("y"),
+                radius=sql.Placeholder("radius"),
+                extra_where=sql.SQL(self.extra_where),
+                temp_predicate=tmp_bounds,
+                limit=sql.SQL("LIMIT {}").format(self.limit and sql.Placeholder("limit") or sql.SQL("ALL"))
             )
         else:
-            sql = """
-                SELECT {}, ST_Distance({}, ST_GeomFromText('POINT(%s %s)', 28992)) AS distance
-                FROM {}
-                WHERE ST_DWithin({}, ST_GeomFromText('POINT(%s %s)', 28992), %s) {}
-            """.format(
-                self.fields, self.meta["geofield"], table, self.meta["geofield"], self.extra_where
+            stmt = sql.SQL("""
+                SELECT {fields},
+                    ST_Distance({geo_field},
+                    ST_GeomFromText('POINT({x} {y})', 28992)) AS distance
+                FROM {schema}.{table_name}
+                WHERE ST_DWithin({geo_field}, ST_GeomFromText('POINT({x} {y})', 28992), {radius}) {extra_where} {temp_predicate} ORDER BY distance {limit}
+            """).format(
+                fields=sql.SQL(", ").join(map(sql.SQL, self.fields.split(","))), # TODO: Properly escape SQL in self.fields string
```

Comment:
Would this work?

```suggestion
                fields=sql.SQL(", ").join(map(sql.Identifier, self.fields.split(","))),
```

Reply:
dit zou ik graag willen maar omdat de waarden in self.fields dus van alles kunnen zijn heb ik die TODO erbij gezet en voor nu ff SQL() instances ervan gemaakt